### ISSUE TITLE
Non-zero status code returned while running Concat node. Name:'Concat_582'. Axis 0 has mismatched dimensions of 5 and 24

### ISSUE THREAD

**Issue by briliantnugraha** [microsoft/onnxruntime#4641]
Hi guys,
I have encountered a strange issue here, where the pytorch-ONNX's output (and output shape) is fine. But I encountered an error as described in my title when I want to use the converted model (onnx) in onnxruntime. FYI, the detectron2 model is working fine in Pytorch. Could anyone help with it? T,T

When I debug it in Spyder, I could see that the first dimension/dim-0 and the output from Pytorch and ONNX seems correct (Picture 3), i.e. inputs are (24,1) and (24,4), and the concatenated output with dim-0 is (24,5). However, when I am trying to feed proposals and features in onnxruntime, the error as in the title (Picture 1) came.
- - - - 
### Picture 1 (ONNXRuntime)
![image](https://user-images.githubusercontent.com/9381132/88624192-1fe7ff80-d0d9-11ea-937d-5f59e4c50589.png)
- - - - 

When I traced the error from ONNXRuntime->Netron->onnx export->Pytorch. I see that the error pointed to 'Concat_582' (**Picture 1**). Then I checked in Netron (**Picture 2**), We could see that the concat node has two inputs, which are **613** and **609**. Then I traced it back to the torch.onnx.export with verbose=True (Picture 3) and check those nodes **(the blue mark ones)**, we could see that the shape is correct for dim=0.
- - - - 
### Picture 2 (Netron)
![image](https://user-images.githubusercontent.com/9381132/88624256-3b530a80-d0d9-11ea-8058-433602335c72.png)
- - - - 
### Picture 3 (Spyder debug, please see the blue marks)
![image](https://user-images.githubusercontent.com/9381132/88624276-4017be80-d0d9-11ea-8ac6-25d6a8363e60.png)
- - - - 

If we traced it to the code as in [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/poolers.py#L75-L77), we could see that it will try to fit the size w.r.t the tensor. Thus, ideally, the result should be the same size (24). Then the concatenation as in [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/poolers.py#L78) should be fine. FYI, the line is not matching in the Picture 3 due to my debugging scripts, but the links that I pointed out are of the same source.

Code to reproduce the error
- To check onnx.export: [Use detectron2, and save only the Roi heads part](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/meta_arch/rcnn.py#L206). You are gonna need to tweak the I/O a bit so that the 'instances conversion error' with [torch.onnx.export](https://pytorch.org/docs/stable/onnx.html) does not happen.
- To check the error in onnxruntime , [please use this files](https://drive.google.com/file/d/1ZYIsGjJsxjkKYzzhQJzwHnk_cQjjAsqa/view?usp=sharing) with the codes below.

```
import onnxruntime as rt
import numpy as np, cv2, os, onnx
from onnx import helper
from onnx import TensorProto
import torch
from PIL import Image
from time import time


print('a. Define input...')
torchpath = r'new_proposals.pt'
inptorch = torch.load(torchpath)

print('b. Define model paths...')
pathmodel = 'detectron2_2.onnx' # put your frozen onnx model here

print('c. Check the model...')
model = onnx.load(pathmodel)
onnx.checker.check_model(model)


print('d. Define session onnxruntime...')
start = time()
sess = rt.InferenceSession(pathmodel)
end = time()
print('Loadtime of onnxruntime: {:.2f}s'.format(end-start))


print('\n\nCheck Input list: ')
[print(i, a.name, a.shape, a.type) for i,a in enumerate(sess.get_inputs())]

#ORT inputs
input0 = sess.get_inputs()[0].name
input1 = sess.get_inputs()[1].name
input2 = sess.get_inputs()[2].name
input3 = sess.get_inputs()[3].name
input4 = sess.get_inputs()[4].name
input5 = sess.get_inputs()[5].name
input6 = sess.get_inputs()[6].name
input7 = sess.get_inputs()[7].name

#Pytorch-Detectron2 inputs for Roi heads
inp0 = inptorch[0]['p2'].cpu().numpy()
inp1 = inptorch[0]['p3'].cpu().numpy()
inp2 = inptorch[0]['p4'].cpu().numpy()
inp3 = inptorch[0]['p5'].cpu().numpy()
inp4 = inptorch[0]['p6'].cpu().numpy()
inp5 = inptorch[1][0].cpu().numpy()
inp6 = inptorch[1][1].cpu().numpy()
inp7 = inptorch[1][2].cpu().numpy()

print('-------\nNumpy input0:', inp0.shape, inp0.dtype)
print('Numpy input1:', inp1.shape, inp1.dtype)
print('Numpy input2:', inp2.shape, inp2.dtype)
print('Numpy input3:', inp3.shape, inp3.dtype)
print('Numpy input4:', inp4.shape, inp4.dtype)
print('Numpy input5:', inp5.shape, inp5.dtype)
print('Numpy input6:', inp6.shape, inp6.dtype)
print('Numpy input7:', inp7.shape, inp7.dtype)


print('Test inference...\n\n')
# This will give the erro as in the Picture 1
outputs = sess.run(None, {input0: inp0, input1: inp1, input2: inp2, \
                                          input3: inp3, input4: inp4, input5: inp5, \
                                          input6: inp6, input7: inp7})#[0]
```

System info:
- OS Platform and Distribution: Windows 10 ver1909
- ONNX Runtime installed from (source or binary): Binary
- ONNX Runtime version: 1.3.0
- Python version: 3.6.5
- Visual Studio version (if applicable): VS 2017 Community Edition
- CUDA/cuDNN version: 10.1/7.6.5
- GPU model and memory: GTX 1080 8GB
[Urgent] If possible, please help me to solve it within this week, your help will be much appreciated.

**Comment by skottmckay** [comment]
I ran the model in the zip using your python code (Windows 10, onnxruntime cpu build version 1.3 and 1.4) and there were no errors so I'm not able to reproduce the issue.

**Comment by briliantnugraha** [comment]
Hi @skottmckay , seems that it is an issue in onnxruntime-gpu 1.3.0 with Cuda 10.1. Using your suggestion solves my problem, thanks a lot!

**Comment by solarflarefx** [comment]
@briliantnugraha How did you export from detectron2 to ONNX without the 'instances conversion error'?  You mentioned tweaking the I/O

**Comment by eklavya-eg** [comment]
can you help me with this
![Screenshot 2024-02-23 005705](https://github.com/microsoft/onnxruntime/assets/129644400/66fc4795-b38d-4f2a-8a87-c80ed4be468f)


Fail: [ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while running Concat node. Name:'/model.18/Concat' Status Message: concat.cc:157 onnxruntime::ConcatBase::PrepareForCompute Non concat axis dimensions must match: Axis 2 has mismatched dimensions of 63 and 64
